src/ledger/InMemoryLedgerTxn.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "ledger/InMemoryLedgerTxn.h"
+#include "ledger/LedgerTxnImpl.h"
+
+namespace stellar
+{
+
+InMemoryLedgerTxn::InMemoryLedgerTxn(AbstractLedgerTxnParent& parent,
+                                     Database& db)
+    : LedgerTxn(parent), mDb(db)
+{
+}
+
+InMemoryLedgerTxn::~InMemoryLedgerTxn()
+{
+}
+
+void
+InMemoryLedgerTxn::addChild(AbstractLedgerTxn& child)
+{
+    if (mTransaction)
+    {
+        throw std::runtime_error(
+            "Adding child to already-open InMemoryLedgerTxn");
+    }
+    LedgerTxn::addChild(child);
+    mTransaction = std::make_unique<soci::transaction>(mDb.getSession());
+}
+
+void
+InMemoryLedgerTxn::commitChild(EntryIterator iter, LedgerTxnConsistency cons)
```

Comment:
It's probably worthwhile to be more careful here and in `rollbackChild` about exception safety. For example, if `LedgerTxn::commitChild(...)` succeeds but `mTransaction->commit()` throws then this will reach an inconsistent state. Looking at `LedgerTxnRoot` might give you a model to work from.

Reply:
and by model, what we end up doing is just `printErrorAndAbort` if this happens - but I agree it's better than throwing exceptions

Reply:
Fixed.